<launch>

	<node pkg="tf2_ros" type="static_transform_publisher" name="map2odom_broadcaster" args="0.2 0.2 0  0 0 0 map odom" />
	<node pkg="tf2_ros" type="static_transform_publisher" name="base2camera_broadcaster" args="0.1 0 0.05  0 0 0 base_link camera" />

	<node name="map_server" pkg="map_server" type="map_server" output="screen" args="$(find mynavigation)/config/map_1.yaml"/>

	
	<group if="$(arg video)">
		<node name="camera" pkg="cv_camera" type="cv_camera_node" output="screen" >
			<!-- Use command "v4l2-ctl -l" to find defaults for a camera, then adjust from there -->
			<param name="camera_info_url" value="$(find mynavigation)/config/logitech_C270_calibration.yml" type="string" />
			
			<param name="cv_cap_prop_brightness" value="128" type="double" />
			<param name="cv_cap_prop_contrast" value="32" type="double" />
			<param name="cv_cap_prop_saturation" value="32" type="double" />
		</node>

		<node name="proc" ns="camera" pkg="image_proc" type="image_proc" output="screen" />

		<!-- Copied/modified from https://github.com/AprilRobotics/apriltag_ros/blob/master/apriltag_ros/launch/continuous_detection.launch -->
		<!-- <include file="$(find apriltag_ros)/launch/continuous_detection.launch" /> -->
		<node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag_ros_continuous_node" output="screen">
			<!-- Remap topics from those used in code to those on the ROS network (?) -->
			<remap from="image_rect" to="camera/image_rect" />
			<remap from="camera_info" to="camera/camera_info" />

			<param name="tag_family" value="tag36h11" type="string" />
			<param name="camera_frame" value="camera" type="string" />
			<param name="publish_tag_detections_image" type="bool" value="false" />
			<param name="queue_size" type="int" value="1" />

			<!-- Load tag ids and size -->
			<rosparam command="load" file="$(find mynavigation)/config/tags.yaml"/>
		</node>
	</group>

	<group if="$(arg navigation)">
		<node pkg="mynavigation" type="mynavigation_node" name="mynavigation_node" output="screen"/>

		<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
			<rosparam file="$(find mynavigation)/config/costmap_common_params.yaml" command="load" ns="global_costmap" /> 
			<rosparam file="$(find mynavigation)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
			<rosparam file="$(find mynavigation)/config/local_costmap_params.yaml" command="load" />
			<rosparam file="$(find mynavigation)/config/global_costmap_params.yaml" command="load" /> 
			<rosparam file="$(find mynavigation)/config/move_base_params.yaml" command="load" /> 
			<rosparam file="$(find mynavigation)/config/base_local_planner_params.yaml" command="load" />
		</node>
	</group>

</launch>
